Generate homebrew formula during imgpkg release

DennisDenuto opened the pull request:
- The github action will commit to carvel-homebrew repo with the updated
formula

Authored-by: Dennis Leon <leonde@vmware.com>

DennisDenuto committed:
Generate homebrew formula during imgpkg release

- The github action will commit to carvel-homebrew repo with the updated
formula

Authored-by: Dennis Leon <leonde@vmware.com>

DennisDenuto commented:
- [x] stop using tags and use commit shas 
- [x] consider inlining the git commit + push action since it gets access to a TOKEN. 

An example of what the homebrew Formula looks like. see: https://github.com/DennisDenuto/homebrew-carvel/pull/2

Note:

In addition to the existing assets that are uploaded (binary files) This will also upload compressed archives "zip" of said binary files. (The homebrew Formula will download and extract the zip files instead of referencing the binary files)


# Prior to merging in:

This github action requires a PAT that has write access to the `vmware-tanzu/homebrew-carvel` repo

Under imgpkg's settings / Secrets / Repository Secrets -->
```
HOMEBREW_TOKEN = PAT
```

DennisDenuto committed:
Inline github action git commit+push

Authored-by: Dennis Leon <leonde@vmware.com>

DennisDenuto committed:
split homebrew to separate job

Authored-by: Dennis Leon <leonde@vmware.com>

DennisDenuto commented:
Going to pivot and change away from goreleaser generating the homebrew file and instead using bash.

Main reason being that goreleaser relies on 'zip' artifacts being uploaded to use in the homebrew file.

In addition to also uploading binary files, having zip equivalents will add complexity down-the-line when we decide to sign and verify our distribution assets.

DennisDenuto committed:
Use bash instead of goreleaser to generate homebrew formula

- this avoids having to upload a 'zip' file since generating homebrew
formulas with goreleaser requires an archive and doesn't allow providing
standalone binaries

Authored-by: Dennis Leon <leonde@vmware.com>

cppforlife commented on .github/workflows/release.yml:
```diff
@@ -97,3 +94,113 @@ jobs:
+              sha256 "$IMGPKG_DARWIN_AMD64"
+            end
+
+            elsif OS.linux?
```
this elif doesnt look right...

DennisDenuto replied:
hmmm how so?

https://github.com/vmware-tanzu/homebrew-carvel/blob/develop/imgpkg.rb#L9

DennisDenuto commented:
This PR is going to change to instead update a metadata file in carvel-release-scripts repo

See the corresponding PR in that repo here: https://github.com/vmware-tanzu/carvel-release-scripts/pull/1

Moving this PR into draft for now...

DennisDenuto committed:
Update carvel-release-scripts with metadata file

- carvel-release-scripts will use the release yaml to generate the
associated assets (such as the homebrew formula) and is responsible to
updating the various distribution channels from it.
- Add ssh-key when checking out carvel-release-scripts
- Fix typo in repo

Authored-by: Dennis Leon <leonde@vmware.com>

DennisDenuto commented:
@cppforlife This is ready for a review and ready to merge. I have added the deploy keys to imgpkg and carvel-release-scripts.

Tested on my fork of imgpkg. Resulted in the following commit made to carvel-release-scripts: https://github.com/vmware-tanzu/carvel-release-scripts/commit/9aa30e4650414c53ae215410c3c648f74bab359a

DennisDenuto commented:
Merging as discussed with dk
